core/src/main/java/org/jdbi/v3/core/statement/SqlStatements.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.statement;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.jdbi.v3.core.JdbiConfig;
+import org.jdbi.v3.core.SqlStatement;
+import org.jdbi.v3.core.StatementContext;
+import org.jdbi.v3.core.TimingCollector;
+import org.jdbi.v3.core.rewriter.ColonPrefixStatementRewriter;
+import org.jdbi.v3.core.rewriter.StatementRewriter;
+
+public final class SqlStatements implements JdbiConfig<SqlStatements> {
+
+    private final Map<String, Object> attributes;
+    private volatile StatementRewriter statementRewriter;
```

Comment:
Agreed, I'll add this to `JdbiConfig` in the form of `implementers of this interface should`, and then make sure it's enforced in all our own config classes.